src/main/thrift/parquet.thrift

```diff
@@ -658,6 +658,74 @@ struct ColumnMetaData {
    * This information can be used to determine if all data pages are
    * dictionary encoded for example **/
   13: optional list<PageEncodingStats> encoding_stats;
+
+  /** Byte offset from beginning of file to bloom filter data. The bloom filters
+   * data of columns together is stored before the start of row group wihch describe.**/
+  14: optional i64 bloom_filter_offset;
+}
+
+/**
+  * Block based algorithm type annotation.
+  */
+struct BlockAlgorithm {
+}
+
+/**
+  * Definition of bloom filter algorithm.
+  * In order for farward compatibility, we use union to replace enum.
+  */
+union BloomFilterAlgorithm {
+  /** Block based bloom filter. 
+   * The bloom filter bitset is separated into tiny bucket as tiny bloom
+   * filter, the high 32 bits hash value is used to select bucket, and
+   * lower 32 bits hash values are used to set bits in tiny bloom filter.
+   * See “Cache-, Hash- and Space-Efficient Bloom Filters”. Specifically,
+   * one tiny bloom filter contains eight 32-bit words (4 bytes stored in
+   * little endian), and the algorithm set one bit in each 32-bit word.
+   *
+   * In order to set bits in bucket, the algorithm need 8 SALT values
+   * (0x47b6137bU, 0x44974d91U, 0x8824ad5bU, 0xa2b7289dU, 0x705495c7U,
+   * 0x2df1424bU, 0x9efc4947U, 0x5c6bfb31U) to calculate index with formular:
+   *                  index[i] = (hash * SALT[i]) >> 27
+   **/
+   1: BlockAlgorithm BLOCK;
+}
+
+/**
+  * Hash strategy type annotation.
+  */
+struct Murmur3 {
+}
+
+/** 
+ * Definition for hash function used to compute hash of column value.
```

Comment:
I would reword this as "The hash function used in the Bloom filter. This function takes the hash of a column value using the plain encoding."

You can leave off the bit about union vs. enum, both here and above.